### PR TITLE
DE-18184-POC-Call-a-Trigger-and-execution

### PR DESCRIPTION
**Pull request by saurabh-saoji-incontact**

### CODE REVIEW

File: src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php [hunk 1]
```diff
@@ -203,6 +203,47 @@ public function expectPlatformAuthorizedRequestFail(
     }
 
 
+    /**
+     * @param mixed[] $responseBody
+     * @param mixed[] $requestOptions
+     */
+    public function expectPlatformAuthorizedRequestDfo3(
```
Comment (jakubkulhanek) [inline src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php:210]: expectDfo3PlatformAuthorizedRequest

File: src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php [hunk 2]
```diff
@@ -203,6 +203,47 @@ public function expectPlatformAuthorizedRequestFail(
...
+            $requestOptions
+        );
+    }
+
+
+    /**
+     * @param mixed[] $responseBody
+     * @param mixed[] $requestOptions
+     */
+    public function expectPlatformAuthorizedRequestDfo3Fail(
```
Comment (jakubkulhanek) [inline src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php:233]: expectDfo3PlatformAuthorizedRequestFail

**Review by jakubkulhanek — APPROVED** [review]

File: src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php [hunk 3]
```diff
@@ -355,4 +396,7 @@ abstract protected function getTempDirectory(): string;
 
 
     abstract protected function getPlatformApiHost(): string;
+
+
+    abstract protected function getPlatformApiHostDfo3(): string;
```
Comment (dominikvoda) [inline src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php:401]: this is BC break FYI, you need to increase major version for new tag of this repository

**Review by dominikvoda — APPROVED** [review]

File: src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php (hunk 1 above)
Comment (saurabh-saoji-incontact) [reply src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php]: done

File: src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php (hunk 2 above)
Comment (saurabh-saoji-incontact) [reply src/MockeryTools/PseudoIntegration/PseudoIntegrationTestCase.php]: done

**Pull request merged**